### DISCUSSION TITLE
Compiling C++20 to use <format>header in windows-runner

### DISCUSSION THREAD

**Discussion by Sunchock** [actions/virtual-environments#7640] [Image configuration]
Hi,

I have wrote a GitHub Action to compile my project with the `windows-latest` runner but it shows me the following error :
```
src/Window.cpp:2:10: fatal error: format: No such file or directory
    2 | #include <format>
      |          ^~~~~~~~
compilation terminated.
make: *** [<builtin>: src/Window.o] Error 1
Error: Process completed with exit code 2.
```
After checking [cpprerefence](https://en.cppreference.com/w/cpp/utility/format/format), <format> header is included in `g++ 13`
I know there is the `fmt` library to replace <format> header but I would like to know if it's possible to use the standard one with the runner.

So i would like to verify if the runner-image uses `g++ 13` and I checked the image details. Actually `windows-latest` is [win22/20230508.3](https://github.com/actions/runner-images/blob/releases/win22/20230508/images/win/Windows2022-Readme.md) and uses `MinGW-w64 11.2.0` but the [official changelogs for MinGW-w64](https://www.mingw-w64.org/changelog/) don't have the `11.2.0` version so it's confusing me.

I added `g++ --version` to my action to get more infos.
```
g++.exe (MinGW-W64 x86_64-posix-seh, built by Brecht Sanders) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
```
So I supposed that `MinGW-w64 11.2.0` version is for g++ in reality and not correspond to the MinGW-w64 itself, maybe a little confusion here ?

Did I miss something ? How to configure runner to get <format> header pls ?

**Comment by ilia-shipitsin** [comment]
are you able to run `g++ 13` locally (on your computer) ? how did you install it ?

**Reply by Sunchock** [reply]
Yeah I'm able to do that, I installed it using mingw64 and adding the g++ folder to my PATH environment variable so I can call it from my base cmd.

**Comment by ilia-shipitsin** [comment]
`MinGW-w64 11.2.0` installed from choco gallery: https://community.chocolatey.org/packages/mingw#versionhistory

**Comment by vpolikarpov-akvelon** [comment]
Hey @Sunchock. You are right, `11.2.0` is not a version of MinGW-w64 itself, but the version of bundled GCC package.

Here is the release that is installed in `windows-latest` image: https://github.com/brechtsanders/winlibs_mingw/releases/tag/11.2.0-12.0.1-9.0.0-r1. It includes GCC 11.2.0 and Clang 12.0.1.

But according to [this page](https://en.cppreference.com/w/cpp/compiler_support), text formatting has been implemented only in GCC 13 and Clang 14. So looks like this feature is not supported by github-hosted runners indeed.

**Reply by Sunchock** [reply]
Hey, thanks for your response. I saw that winlibs_mingw now supports GCC 13 in latest versions. Does this mean github-hosted is now using this version please?

**Reply by erik-bershel** [reply]
Hey @Sunchock!
Sorry to say it, but `GCC` `12.2` is the latest [available](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md?plain=1#L76) in `windows-latest`.
